recipes/backward-cpp/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+       "stack_details_dw": [True, False],
+       "stack_details_bfd": [True, False],
+       "stack_details_dwarf": [True, False],
+       "shared": [True, False],
+       "fPIC": [True, False]
+    }
+    default_options = {
+       "stack_walking_unwind": True,
+       "stack_walking_backtrace": False,
+       "stack_details_auto_detect": False,
+       "stack_details_dw": False,
+       "stack_details_bfd": False,
+       "stack_details_dwarf": True,
+       "stack_details_backtrace_symbol": False,
+       "shared": True,
+       "fPIC": True
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+
+    def config_options(self):     
+          if self.settings.os == "Macos":
+            del self.options.stack_details_dw
+            del self.options.stack_details_bfd
+            del self.options.stack_details_dwarf
+    
+    def _check_options(self, options):
+        sum = 0
+        for name, value in self.options.values.as_list():
+            if value and name in options:
+                sum = sum + 1
+        if sum != 1:
+            return False
+        return True
+
+    def configure(self):
+        if self.settings.os not in ["Linux", "Macos", "Android"]:
+            raise ConanInvalidConfiguration("upstream backward-cpp v{0} is not \
+                supported in {1}.".format(self.version, self.settings.os))
+        # windows implementation only available in upstream master branch
+        
+        if not self._check_options(['stack_walking_unwind', 'stack_walking_backtrace']):
+            raise ConanInvalidConfiguration("Please select stack_walking_unwind"
+                                            " or stack_walking_backtrace.")
+
+        if self.settings.os in ["Linux", "Android"]:
+            if not self._check_options(['stack_details_dw', \
+                                        'stack_details_bfd', \
+                                        'stack_details_dwarf', \
+                                        'stack_details_backtrace_symbol']):
+                raise ConanInvalidConfiguration("Please select stack_details_dw"\
+                    ", stack_details_bfd, stack_details_dwarf or " \
+                    "stack_details_backtrace_symbol.")
+    
+        if self.settings.os == "Macos":            
+            if not self._check_options(['stack_details_backtrace_symbol']):
+                raise ConanInvalidConfiguration("Please select stack_details_backtrace_symbol.")
+
+    def requirements(self):
+        if self.settings.os in ["Linux", "Android"] and \
+           self.options.stack_details_dwarf:
+            self.requires("libdwarf/20191104")
+    
+    def system_requirements(self):
+        required_package = None
+        if self.settings.os == "Linux":
+            if self.options.stack_details_dw:
+                required_package = "libdw-dev"
+            if self.options.stack_details_bfd:
+                required_package = "binutils-dev"
```

Comment:
fragile, you need to check other distros too (centos, fedora, debian, arch, ...)